lib/ash/resource/transformers/require_alphanumeric_field_names.ex

```diff
@@ -0,0 +1,49 @@
+defmodule Ash.Resource.Transformers.RequireAlphanumericFieldNames do
```

Comment:
The module name does not represent the truth of accepted values (underscore is not alphanumeric). A better name suggestion is very welcome :)

Reply:
Perhaps `ValidateFieldNames`?